### ISSUE TITLE
[Bug]:  Dialog, popover trigger should have `aria-haspopup` attribute

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#33478]
### Component

Popover

### Package version

9.56.6

### React version

18.3.1

### Environment

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (22) x64 Intel(R) Core(TM) Ultra 7 165H
    Memory: 33.96 GB / 63.64 GB
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
```

### Current Behavior

the trigger element which opens dialog has aria attribute `aria-expanded="false"`

https://react.fluentui.dev/?path=/docs/components-popover--docs#trapping-focus

### Expected Behavior

The element should have attribute `aria-haspopup="dialog"`

Current Narration of `aria-haspopup="dialog"` is following:
NVDA:  ... button  opens dialog  
JAWS: ... button Has Popup dialog
Narrator:  ... Button Has Popup
VoiceOver: ... dialog popup button

**would be good to consider for fix as well dialog component**
For dialog: https://react.fluentui.dev/?path=/docs/components-dialog--docs#default I cannot see any attribute added.

**Be careful**: we should add `aria-haspopup` only for popover which opens dialog, not others popover...
### Reproduction

https://stackblitz.com/edit/zpkcyref?file=src%2Fexample.tsx

### Steps to reproduce

1. Go to stackblitz
2. verify aria attribute for the button

### Are you reporting an Accessibility issue?

None

### Suggested severity

High - No workaround

### Products/sites affected

teams

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by miroslavstastny** [comment]
@bsunderhus could you please check this one?

**Comment by bsunderhus** [comment]
If I'm not mistaken, there was already a discussion about including or not the `aria-haspopup` attribute before 🤔, I think the previous conclusion was to not include it?! @smhigley , do you have any idea about this topic?
